openstack/networking/v2/extensions/rbac/requests.go

```diff
@@ -0,0 +1,39 @@
+package rbac
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToRbacCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts represents options used to create a rbac_policy.
+type CreateOpts struct {
+	Action       string `json:"action"`
```

Comment:
It looks like there are only two valid options for "action": `access_as_external` and `access_as_shared`. We can create a dedicated type for these. For example:

```go
type PolicyAction string

const (
  ActionAccessExternal PolicyAction = "access_as_internal"
  ActionAccessShared PolicyAction = "access_as_shared"
)
```

And then this CreateOpts field can be defined as:

```go
Action PolicyAction `json:"action" required:"true"`
```

See the compute flavors [requests.go](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/flavors/requests.go) file for an example of this.

The type and const names are only examples. You're more than welcome to use different names if you have some in mind.

Reply:
Thank you for your input, incorporated suggested changes in the latest commit.